src/Tribe/Main.php

```diff
@@ -611,6 +613,7 @@ public function bind_implementations() {
 		tribe_singleton( 'context', 'Tribe__Context' );
 		tribe_singleton( 'post-transient', 'Tribe__Post_Transient' );
 		tribe_singleton( 'db', 'Tribe__Db' );
+		tribe_singleton( 'db-lock', DB_Lock::class );
```

Comment:
Shouldn't we also start registering `DB_Lock::class` as the singleton name?

Reply:
I'm not sure that is the best approach. 
I would rather opt for some kind of auto-wiring in the container component that would understand that, when I say, `tribe_singleton('db-lock', DB_Lock::class)`, then a second registration should be implicitly done to register `DB_Lock` too.

Reply:
I like that solution to the problem.

Reply:
Me too - for :allthethings: singleton?

Reply:
See https://github.com/moderntribe/tribe-common/pull/1392 for a possible implementation of this.